src/samples/index.ts

```diff
@@ -22,23 +22,26 @@ const primitives = {
   number: () => 0,
   number_float: () => 0.0,
   integer: () => 0,
-  boolean: schema => (typeof schema.default === 'boolean' ? schema.default : true),
+  boolean: (schema: SchemaObject): boolean => (typeof schema.default === 'boolean' ? schema.default : true),
 };
 
-const primitive = schema => {
+const primitive = (schema: SchemaObject) => {
   schema = objectify(schema);
   const { type, format } = schema;
 
+  // @ts-expect-error Suppressing `any` errors here because if what we're looking for isn't in `primitives` we throw.
   const fn = primitives[`${type}_${format}`] || primitives[type];
```

Comment:
The error I'm suppressing here:

> Element implicitly has an 'any' type because expression of type '`${string}_${string}`' can't be used to index type '{ string: () => string; string_email: () => string; 'string_date-time': () => string; string_date: () => string; 'string_YYYY-MM-DD': () => string; string_uuid: () => string; string_hostname: () => string; string_ipv4: () => string; ... 4 more ...; boolean: (schema: SchemaObject) => boolean; }'.ts(7053)

Is there a better way to handle this?

Reply:
hard to say without playing with the code directly, here's a related [stack post](https://stackoverflow.com/questions/57086672/element-implicitly-has-an-any-type-because-expression-of-type-string-cant-b). I would just make sure `primitives` is typed properly and if that doesn't work you can always `as` it

Reply:
Was able to fix this by typing `primitives` to the following:

```
Record<string, (arg: void | RMOAS.SchemaObject) => primitive | boolean>
```